src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -52,6 +52,13 @@ protected internal static bool TryParseLogLevel(string value, out LogLevel level
 			}
 		}
 
+		protected bool ParseVerifyServerCert(ConfigurationKeyValue kv)
+		{
+			if (kv == null || string.IsNullOrEmpty(kv.Value)) return DefaultValues.VerifyServerCert;
+
+			return !bool.TryParse(kv.Value, out var value) || value;
```

Comment:
That is why I don't like these "too smart for its own good" Rider refactoring suggestions - for example here it assumes that `DefaultValues.VerifyServerCert` is `true`. If you change `DefaultValues.VerifyServerCert` to `false` then `return !bool.TryParse(kv.Value, out var value) || value;` will be incorrect since it will return `true` when it cannot parse.

Reply:
added a test for the default configuration value

Reply:
Test is good although it doesn't get to `return !bool.TryParse(kv.Value, out var value) || value;` line - the test covers `string.IsNullOrEmpty(kv.Value)` is `true` branch.